salt/modules/zypperpkg.py

```diff
@@ -1356,7 +1356,7 @@ def mod_repo(repo, **kwargs):
     return repo
 
 
-def refresh_db(force=None, root=None):
+def refresh_db(force=None, root=None, **kwargs):
```

Comment:
More freedom to add more in the future. yumpkg only used kwargs.

Reply:
And we have the opportunity to take kwargs from install() and update() and send it to refresh_db() .
It would then automatically pick the important once for refresh and use it.
I did not yet include this into this PR.

Reply:
In general it could hide the reasons of the issue. For example in case of typos in the parameter name. In case of using `kwargs` you can specify `gpgavtoimport`, which makes no sense, but if use the named variable in the signature it will lead to the error. And if I'm not mistaken docstrings usully contains only named variables, the others specified in kwargs not usually listed in the docstrings, but it's not a strict rule I suppose.

Reply:
we ageed to stay with this

Reply:
I would tend to agree with @vzhestkov here, explicit is better than implicit.  We should stray away from utilizing `**kwargs` in modules when we can, as this can cause confusion in errors, etc.... @mcalmer Are you willing to change this over to use explicit keyword arguments?

Reply:
I need to check when i am back at a real computer. May take some days